agent/engine/task_manager.go

```diff
@@ -543,8 +544,24 @@ func (mtask *managedTask) handleEventError(containerChange dockerContainerChange
 	// event.Status is the desired container transition from container's known status
 	// (* -> event.Status)
 	case api.ContainerPulled:
-		// Container's desired transition was to 'PULLED'. A failure to pull might
-		// not be fatal if e.g. the image already exists.
+		// If the agent pull behavior is always or once, we receive the error because
+		// the image pull fails, the task should fail. If we don't fail task here,
+		// then the cached image will probably be used for creating container, and we
+		// don't want to use cached image for both cases.
+		// If the agent pull behavior is never, we won't receive error here, the case
+		// where there is no local image cache will be handled when creating container.
+		if mtask.cfg.AgentPullBehavior == config.AlwaysAgentPullBehavior ||
+			mtask.cfg.AgentPullBehavior == config.OnceAgentPullBehavior {
+			seelog.Criticalf("Managed task [%s]: Error while pulling container %s, task will fail: %v",
```

Comment:
you should probably log the container image here as well.

Reply:
OK